src/effects/RealtimeEffectStateUI.cpp

```diff
@@ -86,6 +99,22 @@ void RealtimeEffectStateUI::Show(AudacityProject& project)
    // Dialog is owned by its parent now!
    mEffectUIHost = dlg.release();
 
+   // Issue 4062, step 3:
+   // For reasons unknown, queueing an event to close the main window really
+   // must happen after the LV2 validator is destroyed, and that must be
+   // done in idle time, outside of any yield to the event loop
+   mValidatorDestroyedSubscription =
+      mEffectUIHost->Subscribe([this](auto){
+         if (mDoClose) {
+            mDoClose = false;
+            // Closing the project during the destructor of another dialog makes
+            // me nervous -- so defer this closing event, too
+            if (mpProjectWindow)
+               DelayedCloseEvent(*mpProjectWindow, mCanVeto);
```

Comment:
So chances are that several Close events are queued. Have you checked that this is safe? Should this be reflected in a message to QA?